packages/formation-react/src/components/MediaLink/MediaLink.jsx

```diff
@@ -0,0 +1,157 @@
+ */
+export const findMediaSite = url =>
+  Object.keys(MEDIA_SITES).find(site =>
+    (url || '').toLowerCase().includes(site),
+  );
+
+/**
+ * Object passed to the media link contents render function
+ * @typedef MediaLink~linkProps
+ * @type {Object.<string, *>}
+ * @property {string} href - link href
+ * @property {string} target - set to '_blank' for external links
+ * @property {string} rel - set to 'noopener noreferrer' for external links
+
+ * @property {string} type - file MIME-type
+ * @property {...*} var_args - Include any extra parameters you want to add to the
+ *  media link, e.g. `className`, `onClick`, etc.
+ */
+/**
+ * Render media link content
+ * @function content
+ * @param {ReactElement} renderedSiteName - rendered site name, e.g.
+ *   `<dfn>on YouTube</dfn>`
+ * @param {MediaLink~props} props - all props passed into the component
+ * @param {MediaLink~linkProps} linkProps - read-only link properties
+ * @return {ReactElement} text with renderedSiteName; or customized as desired
+ */
+/**
+ * MediaLink properties
+ * @typedef MediaLink~props
+ * @type {object.<string, *>}
+ * @param {string} href (required) - url pointing to the video hosting site
+ * @param {string} title (required) - title of the file or video
+ * @param {function} content - Function that allows custom rendering of the link
+ *  content; This function returns JSX
+ * @param {string|ReactElement} icon - Fontawesome icon group + name, e.g.
+ *  "fab fa-youtube", or the icon JSX
+ * @param {boolean} external - Set to true if the link is to an external site;
+ *  setting this adds a blank target (opens a new tab) & no-referrer and opener
+ *  rel attribute
+ * @param {...*} var_args - Include any extra parameters you want to add to the
+ *  video link, e.g. `className`, etc.
+ */
+/**
+ * MediaLink component
+ * @function MediaLink
+ * @param {MediaLink~props}
+ */
+const MediaLink = props => {
+  if (!props.href) {
+    throw new Error('Media links require an href property');
+  }
+
+  const { external = true, title, content, icon = '', ...linkProps } = props;
+  const mediaSite = MEDIA_SITES[findMediaSite(linkProps.href)] || {};
+
+  if (!title && typeof content !== 'function') {
+    throw new Error(
+      'Media links require a title property or custom content function',
+    );
+  }
+
+  // External site
+  if (external) {
+    linkProps.target = '_blank';
+    linkProps.rel = 'noopener noreferrer';
+  }
+
+  const renderedSiteName = mediaSite.name && <dfn>on {mediaSite.name}</dfn>;
+  const renderedSiteType = mediaSite.type || 'site';
+  return (
+    <a {...linkProps}>
+      {typeof icon !== 'string'
+        ? icon
+        : createIcon(icon || mediaSite.icon || 'fas fa-play-circle')}
+      {(typeof content === 'function' &&
+        content({ renderedSiteName, renderedSiteType, linkProps, props })) || (
+        <>
+          View {renderedSiteType} about {title} {renderedSiteName}
```

Comment:
Does it make sense to have default text in here? It doesn't sit quite right with me, but I can't quite put my finger on why that is.